app/lib/analyzer/pana_runner.dart

```diff
@@ -162,6 +166,21 @@ class AnalyzerJobProcessor extends JobProcessor {
     return status;
   }
 
+  bool _isLegacy(Suggestion s) {
+    final isVersionFailed = s.isError &&
+        s.code == SuggestionCode.pubspecDependenciesFailedToResolve &&
+        s.description != null &&
+        s.description.contains('version solving failed');
+    if (!isVersionFailed) {
+      return false;
+    }
+    return s.description.contains('requires SDK version 0.') ||
```

Comment:
Can you add a little comment about exactly what we're trying to compute here and when it applies?